server.R

```diff
@@ -163,6 +163,71 @@ function(input, output, session) {
       colnames(data.gsd) <- c("", "Power (%)")
     return(data.gsd)
   }, striped = T, hover = T, bordered = T, rownames = FALSE, colnames = TRUE, digits = 4)
+  
+  
+  powerGSD <- reactive({
+    # values for tau
+    TAU = unique(sort(c(input$tauGSD, c(seq(0.10, 0.80, 0.10), seq(0.85, 0.95, 0.05), 0.99)*100)))
+    # matrix for results
+    data.gsd = matrix(NA, ncol=4, nrow=length(TAU))
+    for(tauGSD in TAU) {
+      # get critical valus for GSD from package rpact
+      crit = getDesignGroupSequential(
+        kMax = 2,
+        typeOfDesign = input$designSelect, 
+        informationRates = c(tauGSD/100, 1),
+        alpha = input$alphaGSD, 
+        sided = 1)$criticalValues
+      # calculate means for first stage and final stage test statistics
+      mu.t0 = (qnorm(1-input$alphaGSD)+qnorm(input$powerGSD/100)) * sqrt(tauGSD/100)
+      mu.t =  (qnorm(1-input$alphaGSD)+qnorm(input$powerGSD/100)) * (tauGSD/100+(1-tauGSD/100)*(1-value$etaGSD))/sqrt(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)
+      # calculate variance matrix for the test statistics
+      SIG = matrix(c(1, 
+                     sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                     sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                     1), ncol=2)
+      # calculate power for fixed design and GSD (first stage and final stage)
+      power.fix = 1-pnorm(qnorm(1-input$alphaGSD), mean = mu.t0 , sd=1)
+      power.gsd.1 = 1-pnorm(crit[1], mean = mu.t0, sd=1)
+      power.gsd.2 = 1-pmvnorm(lower = c(-Inf, -Inf), upper = c(crit[1], crit[2]), 
+                              mean = c(mu.t0 , mu.t), 
+                              sigma = SIG)[1]
+      data.gsd[which(tauGSD==TAU), 1:4] = c(tauGSD, power.fix*100, power.gsd.1*100, power.gsd.2*100)
+    }
+    return(data.gsd)
+  })
+  
+  output$power_plotGSD <- renderPlot({ # GSD Plot in Tab: Power Information

```

Comment:
I tried to convert this plot to a plotly object so that it is more interactive and colorful. You can simply replace output$power_plotGSD by the code below. And in ui.R, change plotOutput('power_plotGSD') to plotlyOutput('power_plotGSD').

You should be able to view the interactivity of plotly using your local rstudio. If not shown, you could try viewing the app in an internet explorer. Hope it works.

  output$power_plotGSD <- renderPlotly({ # GSD Plot in Tab: Power Information
    data.gsd = powerGSD()
    data_plot <- data.frame(Proportion=rep(data.gsd[, 1], 3), Power=as.vector(data.gsd[, 2:4]),
                       Type=rep(c("Fixed", "GSD (stage 1)", "GSD (overall)"), each=nrow(data.gsd)))
    data_plot$Type <- factor(data_plot$Type, levels=c("Fixed", "GSD (stage 1)", "GSD (overall)"))
    data_plot$Power <- round(data_plot$Power)
    p <- ggplot(data_plot, aes(x = Proportion, y = Power, color = Type)) +
      geom_line(linetype = 1, size = 1) +
      xlab("Proportion (%) of data available") +
      ylab("Power (%)") +
      theme_bw() +
      scale_color_manual(values = c("Fixed" = "#0460A9", "GSD (stage 1)" = "#EC9A1E", "GSD (overall)" = "#8D1F1B")) +
      theme(text = element_text(size = 12, face = "bold"),
            plot.title = element_text(colour = "black", size = 12, face = "bold", hjust = 0),
            legend.title = element_blank(),
            plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
            legend.text = element_text(colour = "black", size = 12, face = "bold")
      )
    ggplotly(p)
  })